src/rpdk/core/contract/resource_client.py

```diff
@@ -98,6 +98,25 @@ def override_properties(document, overrides):
     return document
 
 
+def create_model_with_properties_in_path(src_document, paths):
+    """Prune all properties except given ones from the document.
+
+    This assumes properties will always have an object (dict) as a parent.
+    The function modifies the document in-place, but also returns the document
+    for convenience. (The return value may be ignored.)
```

Comment:
Could we update the method comment?  The modify in-place and return value may be ignored is confusing.